### ISSUE TITLE
False positive in  `unnecessary_operation`

### ISSUE THREAD

**Issue by YoJn** [rust-lang/rust-clippy#5012]
- hi all

```
warning: statement can be reduced
  --> src/utils.rs:33:9
   |
33 | /         &soft_result.push(SoftwareResponse{
34 | |             id: Uuid::new_v4().to_string(),
35 | |             cpe: soft.cpe,
36 | |             installation_id: None,
37 | |             license: soft.license,
38 | |             service_id: soft.service_id,
39 | |         });
   | |___________^
   |
   = note: `#[warn(clippy::unnecessary_operation)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_operation
help: replace it with
   |
33 |         soft_result.push(SoftwareResponse{
34 |             id: Uuid::new_v4().to_string(),
35 |             cpe: soft.cpe,
36 |             installation_id: None,
37 |             license: soft.license,
38 |             service_id: soft.service_id,
 ...
```
- if I remove the semicolon,  the rust compiler thinks this is a return, in fact, this is just an assignment statement

- rustc 1.40.0 (73528e339 2019-12-16)

- clippy 0.0.212 (c8e3cfb 2019-10-28)

**Comment by flip1995** [comment]
Can you share the code where this happens?

**Comment by flip1995** [comment]
I think it's only telling you to remove the `&`, not the semicolon/parentheses/braces, but the span formatting suggests otherwise.

**Comment by YoJn** [comment]
hi @flip1995 ,

You are right!, when I delete the ```&```, it doesn't warning me.  I have a problem with the direction of my thinking.

Thanks for your help
